src/csvcubed/utils/sparql_handler/csvw_state.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from functools import cached_property
+from pathlib import Path
+from typing import Any, Dict, List
+
+import rdflib
+
+from csvcubed.models.csvwtype import CSVWType
+from csvcubed.models.sparqlresults import CatalogMetadataResult
+from csvcubed.utils.sparql_handler.sparql import path_to_file_uri_for_rdflib
+from csvcubed.utils.sparql_handler.sparqlquerymanager import (
+    ask_is_csvw_code_list,
+    ask_is_csvw_qb_dataset,
+    select_csvw_catalog_metadata,
+)
+
+
+@dataclass
+class CsvWState:
+    rdf_graph: rdflib.ConjunctiveGraph
+    csvw_json_path: Path
+
+    def __post_init__(self):
+        self.primary_graph_uri = path_to_file_uri_for_rdflib(self.csvw_json_path)
```

Comment:
Can you make sure to declare this field in the above list? 

```
primary_graph_uri: str = field(init=False)
```

should suffice to make it clear it's not to be part of the `__init__` constructor.